lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -158,16 +144,51 @@ protected function doGetStats()
         );
     }
 
+    private function setDirectory($directory)
+    {
+        if ( ! $this->createPathIfNeeded($directory, 0777)) {
+            throw new \InvalidArgumentException(sprintf(
+                'The directory "%s" does not exist and could not be created.',
+                $directory
+            ));
+        }
+
+        if (function_exists('posix_geteuid')) {
+            $userData = posix_getpwuid(posix_geteuid());
+            $directory .= DIRECTORY_SEPARATOR
```

Comment:
This change seems to create parallel cache dirs (one per user). Looks incorrect to me, although I understand why you tried to go in this direction.
I simply would suggest not sharing a cache dir across different users (seems like a bad idea in first place)